src/app_service/service/transaction/service.nim

```diff
@@ -196,6 +199,7 @@ QtObject:
       error "error: ", errDescription
 
   proc setTrxHistoryResult*(self: Service, historyJSON: string) {.slot.} =
+    self.txCounter -= 1
```

Comment:
is there a chance the counter could go under 0 if networks change and `loadTransactions` is called again? So kind of race condition. I'm asking, because you check `== 0` below, so if it drops below 0 it wouldn't trigger

Reply:
Hmm looks like we could get calls to `setTrxHistoryResult` from tasks launched by a previous call to `loadTransactions`. I believe this would cause signal `SIGNAL_TRANSACTION_LOADING_COMPLETED_FOR_ALL_NETWORKS` to be emitted before it should (though I don't think there's the chance it will not be emitted). We should probably only decrease the counter when `setTrxHistoryResult` is called by tasks matching the last call to `loadTransactions` (or keep a list of the parameters used to launch the tasks for each address, and remove the corresponding one each time the slot is called. When the list for a certain address becomes empty, we emit the signal)

Reply:
and done @jrainville @dlipicar Ive added a Table to make sure that request counter is working for an address at a time.
please check when you can :)